### PR TITLE
Decrypt Function

### PR DESCRIPTION
**Pull request by NatHillardUSDS**
Addresses http://app.clickup.com/t/25qr6et - Create Decryption function to work against incoming data. 

This creates a blob-triggered function that watches "bronze/raw", and when it sees new files, decrypts them and saves them to "bronze/decrypted" with the suffix `.decrypted`. 

Logs from successful run using a single sample file [here](https://portal.azure.com/#blade/WebsitesExtension/FunctionMenuBlade/monitor/resourceId/%2Fsubscriptions%2F7d1e3999-6577-4cd5-b296-f518e5c8e677%2FresourceGroups%2Fprime-ingestion-test%2Fproviders%2FMicrosoft.Web%2Fsites%2Fpitest-functionapp%2Ffunctions%2FDecryptFunction)

### CODE REVIEW

File: src/FunctionApps/PITest_FunctionApp/tests/assets/encrypted.txt [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+wSRQodvrFBanLmE/fNz7vwPEWRUYI+M54okBr9CYzYixVRa+22RnDIvz83qjC2nA
+YCOSQ9ikVw3YtD/eOskffDyvZ8TH6QvCPvcwxxrVJBDlsu1IoUzCHxLwOorlArsx
+V4KK2bryPUnOC6jnMSIhKEa472Ls2QnQAczubUrBy/2EkwSOmNLxjnJU7epStY6J
+Qs4rK+8+l/tGHfbohNdMbeHRME7RGfnQz+XJ+0eLBQOwVqOIvNK8OgmVCV8bgcfO
+Gm6fu7n5odoALkx0X+4BjRlTUnmKCxaDYObycCRI4enQtXv61FYrcHtTHXsEafYU
++Mw04pHKu6LeD5gMpd8IaUeTDqz5+xDKOsJWbg1lF7n5iMOF7y4gd59KUtym6o/S
+YAHUq6ZVfUTZdERma5EKHavYgGdv+1oVgJXGqhnsd4XB44Hggs4deLm8Bh0GVRIz
+eg8Ij6axIvR+u+/2wyOsxM2ULAhQOsDJKNoi1dfolLZiqmGTLAkxKhpORfwZDYmd
+jQ==
+=42tT
```
Comment (bamader) [inline src/FunctionApps/PITest_FunctionApp/tests/assets/encrypted.txt:18]: Do we want to commit this to the repo?

File: workspace.code-workspace [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+      "path": "src/FunctionApps/PHDISmartyStreetsTest"
+    },
+    {
+      "path": "src/FunctionApps/PITest_FunctionApp"
+    }
+  ],
+  "settings": {
+    "debug.internalConsoleOptions": "neverOpen"
+  }
+}
```
Comment (bamader) [inline workspace.code-workspace:13]: Did you mean to commit this?

**Review by bamader — APPROVED** [review]

File: src/FunctionApps/PITest_FunctionApp/tests/assets/encrypted.txt (hunk 1 above)
Comment (JSkeets) [reply src/FunctionApps/PITest_FunctionApp/tests/assets/encrypted.txt:18]: ^ I think this is necessary for the test ? @bamader , obviously Nat will confirm

**Review by JSkeets — APPROVED** [review]
LGTM, agree with Brandon's comment about possible file being included that may not want to be committed.

File: workspace.code-workspace (hunk 2 above)
Comment (NatHillardUSDS) [reply workspace.code-workspace:13]: Yep - I should actually do a brief writeup about this; using a workspace makes it a lot easier to work with and deploy Azure Functions in VSCode as it works better with the Azure extension. I can write this up and add to this PR!

File: src/FunctionApps/PITest_FunctionApp/tests/assets/encrypted.txt (hunk 1 above)
Comment (NatHillardUSDS) [reply src/FunctionApps/PITest_FunctionApp/tests/assets/encrypted.txt:18]: Yep - this is for the test, it's a simple "TEST EICR MESSAGE" message.

File: workspace.code-workspace (hunk 2 above)
Comment (NatHillardUSDS) [reply workspace.code-workspace:13]: (I know Bryan is working on this also, I can make sure we combine notes)

**Comment by bryan-skylight** [comment]
I would prefer to see only the docs relevant to the decrypt function included in this PR, but I don't feel so strongly about it that I'd reject this PR. In general, though, I'd prefer to keep PRs self contained for easier auditability.

**Comment by NatHillardUSDS** [comment]
> I would prefer to see only the docs relevant to the decrypt function included in this PR, but I don't feel so strongly about it that I'd reject this PR. In general, though, I'd prefer to keep PRs self contained for easier auditability.

This is a good point, I can break this out into a separate PR

**Pull request merged**